siteimprove/admin/js/siteimprove.js

```diff
@@ -39,24 +39,36 @@
       this.common();
     },
     common: function (url) {
-      var _si = window._si || [];  
-
-      var getDomCallback = async function () {
-      	var pageWindow = window.open(
-      		url,
-      		"Page Preview",
-      		"width=400,height=500"
-      	);
-      	var promise = new Promise(function (resolve, reject) {
-      		pageWindow.addEventListener(
+      const _si = window._si || [];  
+    
+      const getDomCallback = async function () {
+        const newDiv = document.createElement("div");
+        newDiv.setAttribute("id","div_iframe"); 
+        document.body.appendChild(newDiv);
+        let	iframeString = "<iframe id='domIframe' src="+ url.concat("&si_preview=1") +" style='height:100vh; width:100%'></iframe>";
+        document.getElementById("div_iframe").innerHTML = iframeString.replace(/&amp;/g , "&");
+      	
+      	const promise = new Promise(function (resolve, reject) {
+          const iframe = document.getElementById("domIframe");
+      		iframe.addEventListener(
       			"load",
-      			() => { resolve(pageWindow.document); },
+      			() => { 
+              const newDocument = iframe.contentWindow.document.cloneNode(true);
+              document.getElementById("div_iframe").innerHTML = "";
```

Comment:
And here you could probably also use the `newDiv` variable - which you actually also use below, so I wonder if you even need to clean out the innerHTML if you remove the whole node below?